src/trieNode.ts

```diff
@@ -35,63 +46,66 @@ module.exports = class TrieNode {
    *   - extention - if the node is an extention
    *   - unknown - if something else got borked
    */
-  static getNodeType (node) {
+  static getNodeType(node: Buffer[]): NodeType {
     if (node.length === 17) {
-      return 'branch'
+      return NodeType.Branch
     } else if (node.length === 2) {
       var key = stringToNibbles(node[0])
       if (isTerminator(key)) {
-        return 'leaf'
+        return NodeType.Leaf
       }
 
-      return 'extention'
+      return NodeType.Extension
     }
+    throw new Error('invalid node type')
   }
 
-  static isRawNode (node) {
+  static isRawNode(node: any): boolean {
     return Array.isArray(node) && !Buffer.isBuffer(node)
   }
 
-  get value () {
+  get value(): Buffer {
     return this.getValue()
   }
 
-  set value (v) {
+  set value(v) {
     this.setValue(v)
   }
 
-  get key () {
+  get key(): number[] {
     return this.getKey()
   }
 
-  set key (k) {
+  set key(k) {
     this.setKey(k)
   }
 
-  parseNode (rawNode) {
+  parseNode(rawNode: Buffer[]) {
     this.raw = rawNode
     this.type = TrieNode.getNodeType(rawNode)
   }
```

Comment:
Can `parseNode` now be deprecated or removed?

Reply:
Yeah removed it.